### PR TITLE
remove conditions for Puppetserver <2.2

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -37,11 +37,7 @@
   $dns_alt_names       = []
   $use_srv_records     = false
 
-  # On versions < 3.7.5 the defined function was optional as such strict variables cannot be used
-  # In some edge cases 'domain' fact may be not defined
-  if versioncmp($::puppetversion, '3.7.5') < 0 {
-    $srv_domain = $::domain
-  } elsif defined('$::domain') {
+  if defined('$::domain') {
```
Comment (ekohl) [inline manifests/params.pp]: With [fact](https://github.com/puppetlabs/puppetlabs-stdlib#fact) ```$srv_domain = fact('domain')``` should work.

File: manifests/params.pp [hunk 2]
```diff
@@ -216,7 +212,7 @@
   # Custom puppetmaster
   # needed due to a PUP-4072
   # more information in https://github.com/theforeman/puppet-foreman/commit/5fe3239da0c6fbac76172f61042a69ab3a7eb4e6
-  if versioncmp($::puppetversion, '3.7.5') < 0 or defined('$::puppetmaster') {
+  if defined('$::puppetmaster') {
```
Comment (ekohl) [inline manifests/params.pp]: Can also be ```puppetmaster = fact('puppetmaster')```

**Comment by mmoll** [comment]
should be all fine now :bowtie:

**Comment by mmoll** [comment]
`Error: Evaluation Error: Unknown function: 'fact'. at /etc/puppetlabs/code/modules/puppet/manifests/params.pp:40:6 on node centos-7-x64.c.travis-ci-prod-4.internal` in the beaker tests... I'll change that back...

**Comment by ekohl** [comment]
It looks like it hasn't made it into a release yet so 👍 when this is green again.

**Comment by mmoll** [comment]
💚

**Comment by ekohl** [comment]
Thanks!

File: manifests/params.pp (hunk 2 above)
Comment (alexjfisher) [inline manifests/params.pp]: @mmoll Do you want to remove the PUP-4072 related comments too?

Comment (mmoll) [inline manifests/params.pp]: can do. 👍

**Pull request merged**